components/script/dom/window.rs

```diff
@@ -1414,6 +1448,16 @@ impl Window {
         let doc = self.Document();
         let referrer_policy = referrer_policy.or(doc.get_referrer_policy());
 
+
+        if let Some(fragment) = url.fragment() {
```

Comment:
Let's add a link to https://html.spec.whatwg.org/multipage/#navigating-across-documents and a `// Step 7` comment.